common/short-hex-str/src/lib.rs

```diff
@@ -126,23 +140,25 @@ mod test {
     use std::{str, u8};
 
     #[test]
-    fn test_hex_encode() {
+    fn test_hex_encode() -> Result<(), Error> {
```

Comment:
For tests, you don't need to return the result. Instead, we often just use `unwrap` or `expect` to ensure that an error is not thrown. If one is, the test will panic and fail. For example you could just do:

`hex_encode(&src, &mut actual).unwrap()`

It also keeps the tests cleaner/simpler.